prometheus_hardware_exporter/collector.py

```diff
@@ -531,8 +531,8 @@ def fetch(self) -> List[Payload]:
         """Load ipmi sel entries."""
         sel_entries = self.ipmi_sel.get_sel_entries(self.config.ipmi_sel_interval)
 
-        if not sel_entries:
-            logger.warning("No recent ipmi sel entries to collect.")
+        if sel_entries is None:
```

Comment:
Does None and [] has different meaning here? So that we have to change the syntax here. 
1. There is no description in get_sel_entries function to describe the return value.
2. I prefer to interpret [] and None as the same thing. It's not easy to distinguish them for the following maintainer and make mistake.

Reply:
After we have the time range filter, there are multiple possible situations.

- List with sen_entry.
- Empty list.(Fail on command or no records in time range)

Empty list here means *no records in time range*.
So we miss a way of return to represent *Fail on command*.

Use None and empty maybe the easiest way for this purpose. Also because  we have a mypy linter which make us hard to return `Union` type. We need to do more check on the type.

So the solution can be:

- `Union` return `Union[Exception, List]`
- Add one more return value which represent if the cmd success.